src/go/rpk/pkg/cli/cloud/namespace/list.go

```diff
@@ -0,0 +1,61 @@
+package namespace
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/oauth"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/oauth/providers/auth0"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/publicapi"
+	controlplanev1beta1 "github.com/redpanda-data/redpanda/src/go/rpk/proto/gen/go/redpanda/api/controlplane/v1beta1"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type listResponse struct {
+	Name string `json:"name" yaml:"name"`
+	ID   string `json:"id" yaml:"id"`
+}
+
+func listCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	return &cobra.Command{
+		Use:   "list",
+		Args:  cobra.ExactArgs(0),
+		Short: "List Namespaces in Redpanda Cloud",
+		Run: func(cmd *cobra.Command, _ []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]listResponse{}); ok {
+				out.Exit(h)
+			}
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			authToken, err := oauth.LoadFlow(cmd.Context(), fs, cfg, auth0.NewClient(cfg.DevOverrides()), false)
+			out.MaybeDie(err, "unable to authenticate with Redpanda Cloud: %v", err)
+			cl, err := publicapi.NewClientSet(cmd.Context(), cfg.DevOverrides().PublicAPIURL, authToken)
+			out.MaybeDie(err, "unable to create the public api client: %v", err)
+
+			listed, err := cl.Namespace.ListNamespaces(cmd.Context(), &controlplanev1beta1.ListNamespacesRequest{})
+			out.MaybeDie(err, "unable to list namespaces: %v", err)
+			var res []listResponse
+			for _, n := range listed.Namespaces {
+				if n != nil {
+					res = append(res, listResponse{n.Name, n.Id})
+				}
+			}
+			sort.Slice(res, func(i, j int) bool { return strings.ToLower(res[i].Name) < strings.ToLower(res[j].Name) })
```

Comment:
why ToLower here?

Reply:
Without it, Go will sort the uppercase letter first.

```
data := []string{"A", "b", "D", "c"}
sort.Slice(data, func(i, j int) bool { return data[i] < data[j] })
```

Will return:
```
[A,D,b,c]
```

Reply:
Yes, it is valid. But here I'm sorting the names; they can use upper case and special characters. During the development, I had a case where it was sorted like:

```
NAME                         ID
Nicola's Playground          [redacted]
alena's                      [redacted]
andrew-integration           [redacted]
...
```

So I just did a case insensitive sort here because it looked weird to me.